src/components/FormName.vue

```diff
@@ -0,0 +1,230 @@
+<template>
+  <form
+    class="form-wrapper"
+    @submit.prevent="onSubmit"
+  >
+    <BaseInput
+      type="text"
+      label="first name"
+      placeholder="e.g. Jessica"
+      v-model="firstName"
+      @blur="isFirstNameTouched = true"
+    />
+    <div class="error">{{ firstNameError }}</div>
```

Comment:
Is this `error` element always visible(even if there's no error)?  Maybe it's better to show it conditionally depending on do we have an error or not